components/org.wso2.carbon.identity.application.authenticator.cognito/src/main/java/org/wso2/carbon/identity/application/authenticator/cognito/CognitoOIDCAuthenticator.java

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.wso2.carbon.identity.application.authenticator.cognito;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.oltu.oauth2.client.response.OAuthClientResponse;
+import org.wso2.carbon.identity.application.authentication.framework.config.builder.FileBasedConfigurationBuilder;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.AuthenticatorConfig;
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.exception.LogoutFailedException;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+import org.wso2.carbon.identity.application.authenticator.oidc.OpenIDConnectAuthenticator;
+import org.wso2.carbon.identity.application.common.model.Property;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Authenticator to handle log in ang log out of the AWS Cognito.
+ */
+public class CognitoOIDCAuthenticator extends OpenIDConnectAuthenticator {
+
+    private static final long serialVersionUID = 9058607724358986002L;
+    private static Log log = LogFactory.getLog(CognitoOIDCAuthenticator.class);
+    private String tokenEndpoint;
+    private String oAuthEndpoint;
+    private String logoutEndpoint;
+    private String userInfoEnoint;
+
+    @Override
+    public boolean canHandle(HttpServletRequest request) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Inside CognitoOIDCAuthenticator.canHandle()");
+        }
+        boolean canHandle = super.canHandle(request);
```

Comment:
simply return this at the end of the function. canHandle variable is not required